src/main/java/com/minecolonies/colony/buildings/BuildingBuilder.java

```diff
@@ -123,9 +255,35 @@ public View(ColonyView c, BlockPos l)
          * @return the window of the builder building.
          */
         @NotNull
+        @Override
         public com.blockout.views.Window getWindow()
         {
             return new WindowHutBuilder(this);
         }
+
+        @Override
+        public void deserialize(@NotNull ByteBuf buf)
+        {
+            super.deserialize(buf);
+
+            int size = buf.readInt();
+            neededResources = new HashMap<>();
+
+            for (int i = 0; i < size; i++)
+            {
+                String block = ByteBufUtils.readUTF8String(buf);
+                int amount = buf.readInt();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'amount' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)